File: assets/api/add_values.py
import urllib2
import json
import datetime
import random

# this is a sample python client for tadding multiple data values
# author: Jiri Kadlec
# to check if data values were added, go to worldwater.byu.edu/app and site 'Teva Test' inside Utah Lake.

username = 'admin'
password = 'password'

sourceID = 15
variableID = 43
siteID = 170
methodID = 10
dataValue1 = random.random() * 100
dataValue2 = random.random() * 100
dataValue3 = random.random() * 100

currentDateTime = datetime.datetime.now()
date1 = currentDateTime + datetime.timedelta(minutes=10)
date2 = currentDateTime + datetime.timedelta(minutes=20)
date3 = currentDateTime + datetime.timedelta(minutes=30)

#STEP 1: prepare the uploaded data in JSON format
data = {
    "SiteID": siteID,
    "VariableID": variableID,
    "MethodID": methodID,
    "SourceID": sourceID,
    "values": [(date1.strftime("%Y-%m-%d %H:%M:%S"), dataValue1),
               (date2.strftime("%Y-%m-%d %H:%M:%S"), dataValue2),
               (date3.strftime("%Y-%m-%d %H:%M:%S"), dataValue3)]
}

postdata = json.dumps(data)
uploadURL = 'http://worldwater.byu.edu/app/index.php/default/services/api/values'
req = urllib2.Request(uploadURL)
req.add_header('Content-Type', 'application/json')

try:
    response = urllib2.urlopen(req, postdata)
    print response.read()

except urllib2.HTTPError, e:
    print e.code
    print e.msg
    print e.headers
    print e.fp.read()

File: assets/api/add_values.R
#you must install the packages httr and RJSONIO.
library(httr)
library(RJSONIO)

sourceID = 15
variableID = 43
siteID = 170
methodID = 10
dataValue1 = runif(1, 0, 100)
dataValue2 = runif(1, 0, 100)
dataValue3 = runif(1, 0, 100)

now = Sys.time()
date1 = now + 5 * 60
date2 = now + 10 * 60
date3 = now + 15 * 60

values <- c(dataValue1, dataValue2, dataValue3)
dates <- c(date1, date2, date3)
dates_formatted <- format(dates, "%Y-%m-%d %H:%M:%S")
value_list <- as.matrix(cbind(dates_formatted, values))
colnames(value_list) <- NULL
rownames(value_list) <- NULL

x <- list(
    user = "admin",
    password = "password",
    SiteID = siteID,
    VariableID = variableID,
    MethodID = methodID,
    SourceID = sourceID,
    values = value_list
    )

resp = POST("http://worldwater.byu.edu/app/index.php/default/services/api/values",
         body = RJSONIO::toJSON(x),
         add_headers("Content-Type" = "application/json"),
         verbose()
    )
response_status = content(resp, type="application/json")
print(response_status)
